src/lib/comms/sol-socket-impl-linux.c

```diff
@@ -289,13 +301,13 @@ is_multicast(enum sol_network_family family, const struct sockaddr *sockaddr)
 }
 
 static ssize_t
-sol_socket_linux_sendmsg(struct sol_socket *socket, const void *buf, size_t len,
+sol_socket_linux_sendmsg(struct sol_socket *socket, const struct sol_buffer *buf,
     const struct sol_network_link_addr *cliaddr)
 {
     struct sol_socket_linux *s = (struct sol_socket_linux *)socket;
     uint8_t sockaddr[sizeof(struct sockaddr_in6)] = { };
-    struct iovec iov = { .iov_base = (void *)buf,
-                         .iov_len = len };
+    struct iovec iov = { .iov_base = (void *)buf->data,
+                         .iov_len = buf->used };
```

Comment:
should we check nul_byte here and use `+1`? likely not... but shouldn't hurt if people use the proper flags (or use `NO_NUL_BYTE` for binary protocols)